src/main/java/net/dries007/tfc/common/blocks/BurningLogPileBlock.java

```diff
@@ -0,0 +1,119 @@
+import net.minecraft.util.math.RayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+
+import net.dries007.tfc.common.TFCTags;
+import net.dries007.tfc.common.tileentity.BurningLogPileTileEntity;
+import net.dries007.tfc.common.tileentity.LogPileTileEntity;
+import net.dries007.tfc.util.Helpers;
+
+public class BurningLogPileBlock extends Block implements IForgeBlockProperties
+{
+    public static void tryLightLogPile(World world, BlockPos pos)
+    {
+        LogPileTileEntity pile = Helpers.getTileEntity(world, pos, LogPileTileEntity.class);
+        if (pile != null)
+        {
+            int logs = pile.countLogs();
+            pile.clearContent(); // avoid dumping when onRemove is called
+            world.setBlockAndUpdate(pos, TFCBlocks.BURNING_LOG_PILE.get().defaultBlockState());
+            Helpers.playSound(world, pos, SoundEvents.BLAZE_SHOOT);
+
+            BurningLogPileTileEntity newPile = Helpers.getTileEntity(world, pos, BurningLogPileTileEntity.class);
+            if (newPile != null)
+            {
+                newPile.light(logs);
+                tryLightNearby(world, pos);
+            }
+        }
+    }
+
+    private static boolean isValidCoverBlock(BlockState offsetState, World world, BlockPos pos, Direction side)
+    {
+        if (offsetState.is(TFCTags.Blocks.CHARCOAL_COVER_WHITELIST))// log pile, charcoal pile, this
+        {
+            return true;
+        }
+        return !offsetState.getMaterial().isFlammable() && offsetState.isFaceSturdy(world, pos, side);
+    }
+
+    private static void tryLightNearby(World world, BlockPos pos)
+    {
+        if (world.isClientSide()) return;
+        for (Direction side : Helpers.DIRECTIONS)
+        {
+            final BlockPos offsetPos = pos.relative(side);
+            BlockState offsetState = world.getBlockState(offsetPos);
+            if (isValidCoverBlock(offsetState, world, offsetPos, side.getOpposite()))
+            {
+                if (offsetState.is(TFCBlocks.LOG_PILE.get()))
+                {
+                    tryLightLogPile(world, offsetPos);
+                }
+            }
+            else
+            {
+                world.setBlockAndUpdate(offsetPos, Blocks.FIRE.defaultBlockState());
+            }
+        }
+    }
+    private final ForgeBlockProperties properties;
+
+    public BurningLogPileBlock(ForgeBlockProperties properties)
+    {
+        super(properties.properties());
+        this.properties = properties;
+    }
+
+    @Override
+    public ForgeBlockProperties getForgeProperties()
+    {
+        return properties;
+    }
+
+    @Override
+    @SuppressWarnings("deprecation")
+    public void randomTick(BlockState state, ServerWorld world, BlockPos pos, Random rand)
+    {
+        tryLightNearby(world, pos);
+    }
+
+    @Override
+    public void animateTick(BlockState stateIn, World worldIn, BlockPos pos, Random rand)
```

Comment:
Missing `@OnlyIn(Dist.CLIENT)` as it overrides a method with the same annotation. (Forge disagrees that we should include this here as 'oh no it's a scary coremod'. There's documented cases of not including it breaking reflection, to which the response was 'so what'. But we're already a big scary coremod so I'd rather not leave the fragile thing there.)

IDEA has an option to configure annotations to copy from superclass methods when overriding, suggest adding `OnlyIn` as one of them.

It's also a good reminder that this is a clientside method